### PR TITLE
Stripe checkout to charge credit card

### PR DESCRIPTION
**Pull request by gvszg**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -21,7 +21,14 @@ def create
...
 
     if @user.save
-      handle_invitation      
+      handle_invitation
+      Stripe.api_key = ENV['STRIPE_SECRET_KEY']
+      Stripe::Charge.create(
+        :amount => 999,
+        :currency => "usd",
+        :source => params[:stripeToken],
+        :description => "Sign up charge for #{@user.email}"
```
Comment (tomtomecek) [inline app/controllers/users_controller.rb]: This looks good, except I would change the hash notation to Ruby 1.9+.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -21,7 +21,14 @@ def create
     @user = User.new(user_params)
 
     if @user.save
-      handle_invitation      
+      handle_invitation
```
Comment (tomtomecek) [inline app/controllers/users_controller.rb]: I would handle invitation _after_ charging the card. IF the charge was successful, then you could save the `@user`. I think the workflow isn't 100% correct here.

File: app/views/videos/show.html.haml [hunk 3]
```diff
@@ -2,14 +2,17 @@
   .container
     .row
       .video_large_cover.col-sm-7.col-sm-offset-1
-        = image_tag @video.large_cover_url
+        - if params[:watch]
+          = video_tag "#{@video.video_url}", type: "video/mp4", size: "665x375", controls: true, autoplay: true
```
Comment (tomtomecek) [inline app/views/videos/show.html.haml]: `video_url` should be a string so there shouldn't be a need for an interpolation - `video_tag @video.video_url, ...`

File: app/views/users/new.html.haml [hunk 4]
```diff
@@ -1,23 +1,36 @@
...
-          = hidden_field_tag :invitation_token, @invitation_token
+            %label.control-label.col-sm-2 Expiration
+            .col-sm-3
+              = select_month(Date.today, {add_month_numbers: true}, class: 'form-control', data: {stripe: "exp-month"})
+            .col-sm-2
+              = select_year(Date.today.year, {start_year: Date.today.year, end_year: Date.today.year + 4}, class: 'form-control', data: {stripe: "exp-year"})
         %fieldset.actions.control-group.col-sm-offset-2
           .controls
-            = f.submit("Sign Up", class: "btn btn-default")
+            = f.submit "Sign Up", class: 'btn btn-default'
```
Comment (tomtomecek) [inline app/views/users/new.html.haml]: Mixing single and double quotes looks strange to me, I would keep double quotes.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (gvszg) [inline app/controllers/users_controller.rb]: Is that following modified one ok?

```ruby
def create
  @user = User.new(user_params)
  begin
    Stripe.api_key = ENV['STRIPE_SECRET_KEY']
      Stripe::Charge.create(
        amount: 999,
        currency: "usd",
        source: params[:stripeToken],
        description: "Sign up charge for #{@user.email}"
      )      
    @user.save
    handle_invitation
    AppMailer.delay.send_welcome_email(@user)
    flash[:warning] = "You've registered!"
    redirect_to sign_in_path
  rescue Stripe::CardError => e
    flash.now[:danger] = "Please check your payment information!"
    render :new
  end
end
```

**Pull request merged**